### ISSUE TITLE
Track response payload size and time 

### ISSUE THREAD

**Issue by stephenkao** [CDCgov/prime-reportstream#8291]
## Problem statement

We'd like to track response payload size and time from call to our API so that we can keep track of request trends for our users.  This way, we can loosely if there are significant improvements or regressions of our API timings.

## What you need to know

- App Insights already tracks a `RemoteDependencyData` event that has time so we can look into adding payload size into the event data:

![Screenshot 2023-02-10 at 14.02.52.png](https://images.zenhubusercontent.com/5999b3b38f62dc7798c36942/d858e5b0-25e9-4fce-ab67-459259e40ac1)

- If it's not easy or possible to extend the `RemoteDependencyData` with our custom metrics, we should look into tracking that through our own custom fetcher.

- Front-end App Insights calls are subject to individual network conditions and being blocked by ad blockers.  This is acceptable because we're just looking for general trends.

## Acceptance criteria

- [ ] All ReportStream API calls on the website are tracked in App Insights with response time and payload size

**Comment by stephenkao** [comment]
@chris-kuryak Added this after our chat today.  We can update this or add another ticket to add these metrics to the App Insights dashboard -- let me know!

**Comment by penny-lischer** [comment]
This is not possible without writing a `customDimension` to the `dependencies` table within App Insights. @stephenkao and I felt that we should defer this ticket until we had a centralized fetch mechanism for our API calls. We could then wrap the fetch with an `appInsights.trackEvent` to capture the `header content.length`.
* More about dependencies - https://learn.microsoft.com/en-us/azure/azure-monitor/app/asp-net-dependencies

**Comment by chris-kuryak** [comment]
@stephenkao @penny-lischer Do we need to create a ticket to create a centralized fetch mechanism to unblock this ticket? Or does that already exist? Just checking on next steps forward.

**Comment by stephenkao** [comment]
@chris-kuryak It's essentially the react-query migration since we're split between two fetching mechanisms right now.  Well, three if you include the markdown page rendering 😬

**Comment by chris-kuryak** [comment]
@stephenkao Thanks for the clarity! I'll add the react query final ticket as a dependency.

**Comment by chris-kuryak** [comment]
This is a follow-on feature for Website Analytics. As such, I am taking it out of the Website Analytics MVP epic.
